### DISCUSSION TITLE
SQL 4 CDS SSMS

### DISCUSSION THREAD

**Discussion by CMSMC** [MarkMpn/Sql4Cds#141] [Q&A]
Hi,

I'm having some issues using SQL 4 SQL SSMS Edition. When I install it it works but when I close SSMS and reopen it it stops working. When I run and IS«NSERT/UPDATE query the add-in is no longer triggered and the only symptom I notice I that the FetchXML button gives an error message: Object reference not set to an instance of an object.

I'm using this version of SSMS
![2021-12-21_23h59_44](https://user-images.githubusercontent.com/96502717/147008528-1e078a05-2b98-4bf2-b2ff-091e0fd20052.png)

The Add-in seems to be loaded
![2021-12-22_00h01_09](https://user-images.githubusercontent.com/96502717/147008607-0a267f07-022a-4948-b197-860c1dbebe2c.png)

Any idea of what can be causing such random behaviour?

Thanks in advance for your support.

**Comment by MarkMpn** [comment]
Could you please try running SSMS with logging enabled to see if any further details are available in the log? To capture a log file go to Start > Run and enter `"C:\Program Files (x86)\Microsoft SQL Server Management Studio 18\Common7\IDE\Ssms.exe" -log "C:\Users\<username>\Desktop\ssms.xml"`

**Reply by CMSMC** [reply]
I find these entries:
![image](https://user-images.githubusercontent.com/96502717/147214970-d4cc6152-e4a3-4349-8c3c-11ba4e3f5272.png)
and
![image](https://user-images.githubusercontent.com/96502717/147215034-f07e22b2-ae97-41f2-9575-253899950ad5.png)

but it doesn't seem to be triggered when I run an update/delete query
![image](https://user-images.githubusercontent.com/96502717/147215276-613d27bc-01bf-441c-a6eb-d89987b89be6.png)

**Reply by MarkMpn** [reply]
You also mentioned getting an error message when you click the FetchXML button in the toolbar - does this add any more information to the log file?

**Reply by CMSMC** [reply]
No. And I just tried in another laptop and I face the same problem.
The convert the SQL to M for Power BI works. But the FetchXML doesn't.

**Reply by MarkMpn** [reply]
Could you show me what the error looks like when you click the FetchXML button please?

I'm adding some logging to the next version to help track this down as I can't reproduce it for the moment.

**Reply by MarkMpn** [reply]
Could you try this again with the latest version to see if there's any more details in the log file now please?

**Comment by CMSMC** [comment]
Hi Mark. I've been using this tool from time to time, never very stable but it stoped working. I see the tollbar loaded and PBI M feature is working. When I run an INSERT/UPDATE command it doesn't seem to trigger the extension.
I installed SSMS 19, and the last version of SQL4CDS for SSMS.
Can you help further?

<img width="725" alt="image" src="https://github.com/MarkMpn/Sql4Cds/assets/96502717/0660cc0c-3406-43aa-acc4-5f15151cb4a3">

<img width="381" alt="image" src="https://github.com/MarkMpn/Sql4Cds/assets/96502717/997d39ee-80c4-4d2d-867a-e4da52be93d4">

And I find the error below in the LOG:
<entry>
    <record>1090</record>
    <time>2023/05/19 10:27:25.266</time>
    <type>**Error**</type>
    <source>SQL 4 CDS</source>
    <description>System.NullReferenceException: Object reference not set to an instance of an object.&#x000D;&#x000A;   at MarkMpn.Sql4Cds.Engine.TableSizeCache..ctor(IOrganizationService org, IAttributeMetadataCache metadata) in D:\a\1\s\MarkMpn.Sql4Cds.Engine\TableSizeCache.cs:line 27&#x000D;&#x000A;   at MarkMpn.Sql4Cds.SSMS.CommandBase.GetDataSource()&#x000D;&#x000A;   at MarkMpn.Sql4Cds.SSMS.DmlExecute.OnExecuteQuery(String Guid, Int32 ID, Object CustomIn, Object CustomOut, Boolean&amp; CancelDefault)</description>
</entry>

**Reply by MarkMpn** [reply]
Are there any other SQL 4 CDS messages in the log prior to this NullReferenceException one?

**Comment by CMSMC** [comment]
I'm affraid not. Full log in attach.

Carlos Cruz
________________________________

**Reply by MarkMpn** [reply]
Sorry, the attachment doesn't seem to have come through. Can you attach it to the discussion directly in GitHub please?

**Reply by MarkMpn** [reply]
Thanks, unfortunately I still haven't been able to reproduce this issue. I wonder if there's something about the order of exactly how you interact with SSMS that might be different to how I normally work which means you're getting different results to me. Is it possible to get a screen recording going through the whole process of starting SSMS, logging in, writing/loading the query and executing it in case there's something in there that might be triggering the problem?

**Comment by CMSMC** [comment]
[SSMS.zip](https://github.com/MarkMpn/Sql4Cds/files/11529671/SSMS.zip)
